MAUI/Calendar/customizations.md

```diff
@@ -222,4 +291,352 @@ You can customize the selection cell background and text style in Month, Year, D
+                                TodayHighlightBrush="#0A3A74"
+                                AllowViewNavigation="False">
+            <calendar:SfCalendar.YearView>
+                <calendar:CalendarYearView CellTemplate="{Binding Template}">
+                </calendar:CalendarYearView>
+            </calendar:SfCalendar.YearView>
+            <calendar:SfCalendar.BindingContext>
+                <local:DecadeTemplate/>
+            </calendar:SfCalendar.BindingContext>
+        </calendar:SfCalendar>
+    </Frame>
+</Grid>
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+public class DecadeTemplate
+{
+    private DataTemplate circleTemplate;
+
+    private DataTemplate template;
+
+    public DataTemplate Template
+    {
+        get
+        {
+            return this.template;
+        }
+        set
+        {
+            this.template = value;
+        }
+    }
+
+    public DecadeTemplate()
+    {
+        this.circleTemplate = new DataTemplate(() =>
+        {
+            Grid grid = new Grid();
+
+            Border border = new Border();
+            border.BackgroundColor = Color.FromRgba("#F5F5F5");
+            border.StrokeShape = new RoundRectangle()
+            {
+                CornerRadius = new CornerRadius(25)
+            };
+
+            border.SetBinding(Border.StrokeThicknessProperty, "Date", converter: new DateToStrokeConverter());
+            border.Stroke = Color.FromArgb("#0A3A74");
+
+            Label label = new Label();
+            label.SetBinding(Label.TextProperty, "Date.Year");
+            label.HorizontalOptions = LayoutOptions.Center;
+            label.VerticalOptions = LayoutOptions.Center;
+            label.Padding = new Thickness(2);
+            border.Content = label;
+
+            grid.Add(border);
+            grid.Padding = new Thickness(1);
+
+            return grid;
+        });
+
+        this.template = this.circleTemplate;
+    }
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="Value converter.cs" %}
+
+internal class DateToStrokeConverter : IValueConverter
+{
+    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        var date = value as DateTime?;
+        if (date.HasValue && date.Value.Year == DateTime.Now.Year)
+        {
+            return 2;
+        }
+
+        return 0;
+    }
+
+    public object? ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        return null;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.
```

Comment:
You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

Reply:
Content modified.